src/graph/optimizer/rule/GetEdgesTransformLimitRule.cpp

```diff
@@ -24,59 +24,73 @@ using nebula::graph::Traverse;
 namespace nebula {
 namespace opt {
 
-std::unique_ptr<OptRule> GetEdgesTransformRule::kInstance =
-    std::unique_ptr<GetEdgesTransformRule>(new GetEdgesTransformRule());
+std::unique_ptr<OptRule> GetEdgesTransformLimitRule::kInstance =
+    std::unique_ptr<GetEdgesTransformLimitRule>(new GetEdgesTransformLimitRule());
 
-GetEdgesTransformRule::GetEdgesTransformRule() {
+GetEdgesTransformLimitRule::GetEdgesTransformLimitRule() {
   RuleSet::QueryRules().addRule(this);
 }
 
-const Pattern &GetEdgesTransformRule::pattern() const {
-  static Pattern pattern =
-      Pattern::create({PlanNode::Kind::kAppendVertices, PlanNode::Kind::kLimit},
```

Comment:
Why don't you keep the origin code ?

Reply:
Previously I used to keep the original rule (with an extra build-path condition check) but found the original rule will anyway “short circuit” this new rule, no matter how the order in the CMake list of the two rules, while when the original rule being applied with transform for `MATCH ()-[e:serve]->() RETURN e,e LIMIT 3`(e being returned more than once, or `return e.PropA, properties(e)` ), there is no `LIMIT` info in that pattern, thus it will end up with limit(-1) to lead it cannot be pushed down. Then I found we actually only care about this pattern when 1. there is no edge index; 2. there is a `LIMIT` clause, thus I could actually remove the original short rule, and it worked fine for all regression cases.

Now, with your change on that rule being merged, the `orLimit` is in the pattern, maybe I should revisit to see if I could  leave the two, and add `buildPath` check in the original rule instead of removing them, I have two talks to be prepared for tonight and tomorrow night, I will come back to this later soon.

Reply:
> Ok, and you should format the feature file by `make -C tests fmt` to avoid CI lint error.

I don't know this, too, thanks!!